libs/sdk-ui-loaders/src/dashboard/useDashboardLoader.ts

```diff
@@ -0,0 +1,134 @@
+// (C) 2021 GoodData Corporation
+import { useMemo, useState } from "react";
+import { IDashboardBasePropsForLoader, IDashboardLoadOptions } from "./types";
+import {
+    IClientWorkspaceIdentifiers,
+    useBackendStrict,
+    useCancelablePromise,
+    useWorkspaceStrict,
+} from "@gooddata/sdk-ui";
+import { IDashboardBaseProps } from "@gooddata/sdk-ui-dashboard";
+import isArray from "lodash/isArray";
+import compact from "lodash/compact";
+import { DashboardLoader } from "./dashboardLoader";
+import { DashboardLoadResult, IDashboardLoader, IEmbeddedPlugin } from "./loader";
+
+/**
+ * @alpha
+ */
+export type DashboardLoadStatus = {
+    /**
+     * Status of the dashboard loading.
+     */
+    status: "loading" | "error" | "success";
+
+    /**
+     * Load result - this will be set when status is 'success'. Undefined in all other states.
```

Comment:
This can be codified on the type level, see: https://github.com/gooddata/gooddata-ui-sdk/blob/master/libs/sdk-ui/src/base/react/useCancelablePromise.ts#L24-L62